File: Hafta 2/pset2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

int main(void)
{
    string user = get_string("Text:");
    int n = strlen(user);
    int lettercount = 0;
    for (int i = 0; i < n ; i++)
    {
        if (isalpha(user[i]))
        {
            lettercount += 1;
        }
    }
    int x = strlen(user);
    int wordcount = 1;
    for (int i = 1; i < x; i++)
    {
        if ((isspace(user[i])) && (isalpha(user[i + 1])))
        {
            wordcount += 1;
        }

    }
    int z = strlen(user);
    int sentencecount = 0;
    for (int i = 1; i < x; i++)
    {
        if (user[i] == '.' || user[i] == '?' || user[i] == '!' || user[i] == ':') 
        { 
            sentencecount += 1;
        }

    }
    float averagenumberoflettersper100words = ((float)lettercount * 100) / (float)wordcount;
    float averagenumberofsentencesper100words = ((float)sentencecount * 100) / (float)wordcount;
    float grade = 0.0588 * averagenumberoflettersper100words - 0.296 * averagenumberofsentencesper100words - 15.8;
    if (grade >= 1 && grade <= 16)
    {
        printf("Grade %i\n", (int) round(grade));
    }
    else
    {
        if (grade < 1)
        {
            printf("Before Grade 1\n");
    
        }
    }
    
    {
        if (grade > 16)
        {
            printf("Grade 16+\n");
        
        }
    }
}

File: Hafta 6/pset6/readability.py
from cs50 import get_string

text = get_string("Text: ")

n_words = n_sent = n_let = i = 0
length = len(text)

while i < length:
    if text[i].isalpha():
        n_let += 1
    if (i == 0 and text[i] != " ") or (i != length and text[i] == " " and text[i + 1] != " "):
        n_words += 1
    if text[i] == "." or text[i] == "?" or text[i] == "!":
        n_sent += 1
    i += 1

L = (n_let / n_words) * 100
S = (n_sent / n_words) * 100
index = round(0.0588 * L - 0.296 * S - 15.8)

if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print("Grade 16+")
else:
    print(f"Grade {index}")